src/services/__tests__/pagoPAProxyService.test.ts

```diff
@@ -177,6 +182,24 @@ describe("PagoPAProxyService#getPaymentInfo", () => {
     expect(res.kind).toEqual("IResponseErrorValidation");
   });
 
+  it("getPaymentInfoDefaultDecoder must validate correctly a PaymentProblemJson response", async () => {
+    const a = {
+      detail: PaymentFaultEnum.DOMAIN_UNKNOWN
+    };
+    const validator = getPaymentInfoDefaultDecoder();
+    const res = mockRes();
+    // tslint:disable: no-object-mutation
+    res.body = a;
+    res.status = 500;
+    // tslint:enable: no-object-mutation
+    const validation = await validator(res);
+    if (validation !== undefined) {
```

Comment:
the decoder should never return `undefined` AFAIK

Reply:
like [here](https://github.com/teamdigitale/io-backend/pull/477#discussion_r321669724)